File: backend/models/inventory.js
const mongoose = require('mongoose');

const inventorySchema = mongoose.Schema({
  supplierId: {type: String , require:true},
  supplierName: {type: String , require:true},
  name: {type: String , require:true},
  quantity: {type: String , require:true},
  expireDate: {type: Date , require:true},
  price: {type: String , require:true},
  imagePath : { type: String , require: true}
})

module.exports = mongoose.model('Inventory',inventorySchema);

File: frontend/lib/models/inventory.dart
import 'dart:ffi';

class InventoryModel {
  String? email;
  String? name;
  int? quantity;
  String? batchId;
  String? expireDate;
  int? price;
  String? supplierId;
  String? supplierName;

  InventoryModel({
    required this.email,
    required this.name,
    required this.quantity,
    required this.batchId,
    required this.expireDate,
    required this.price,
    required this.supplierId,
    required this.supplierName,
  });
  factory InventoryModel.fromJson(Map<String, dynamic> json) {
    return InventoryModel(
      email: json['email'],
      name: json['name'],
      quantity: json['quantity'],
      batchId: json['batchId'],
      expireDate: json['expireDate'],
      price: json['price'],
      supplierId: json['supplierId'],
      supplierName: json['supplierName'],
    );
  }
}
